jax/_src/scipy/stats/multinomial.py

```diff
@@ -0,0 +1,34 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import scipy.stats as osp_stats
+from jax import lax
+from jax._src.numpy import lax_numpy as jnp
+from jax._src.numpy.lax_numpy import _promote_args_inexact
+from jax._src.numpy.util import _wraps
+from jax._src.scipy.special import gammaln, xlogy
+
+@_wraps(osp_stats.multinomial.logpmf, update_doc=False)
+def logpmf(x, n, p):
+  """JAX implementation of scipy.stats.multinomial.logpmf."""
+  x, p = _promote_args_inexact("multinomial.logpmf", x, p)
```

Comment:
I'm somewhat concerned with this implementation allowing floating point inputs for `x`. The function is not defined for non-integer `x` values. In numpy, this is enforced by a runtime check that effectively checks `all(x.astype(int) == x)`, but of course we can't do this in JAX because at runtime we may be on GPU or TPU where this kind of error propagation is difficult.

The path chosen here is to silently proceed with the computation even if `x` is invalid. But we have another option: if x has integer dtype, then we *know* that it's valid. So a different option would be to require an integer dtype for `x`. What do you think?

Reply:
Hello sir, I am a novice to programming and I tried to cast x to an integer in this PR for @harryjulian 's branch. May you please review it? Thank you sir!

Reply:
If you're going to be making significant changes here and @harryjulian is no longer working on this branch, it would probably be cleaner to open a new pull request to avoid all this kind of indirection during review.